src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 4;
+    private static final int MAX_NUMBER = 10;
+    private static final int CIRCLE_RADIUS = 10;
+    private static Random random = new Random();
+    private static ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(NUMBER_OF_FIGURES);
+        switch (index) {
+            case 0:
+                int sideOfSquare = random.nextInt(MAX_NUMBER) + 1;
+                return new Square(sideOfSquare,
+                        colorSupplier.getRandomColor());

```

Comment:
For the future, you can move the logic to private method
```suggestion
                return prepareSquare();

```